<?xml version="1.0"?>

<launch>
    <arg name="log_path" default="$(find metacontrol_experiments)/data"/>
    <arg name="record_bags" default="false"/>
    <arg name="bag_store_path" default="$(find metacontrol_experiments)/bags/"/>
    <arg name="store_data_freq" default="2.0"/>
    <arg name="obstacles" default="0"/>
    <arg name="goal_nr" default="1"/>
    <arg name="increase_power" default="0.0"/>
    <arg name="send_laser_error" default="false"/>
    <arg name="nfr_energy" default="0.5"/>
    <arg name="nfr_safety" default="0.5"/>
    <arg name="nav_profile" default=""/>

    <node pkg="metacontrol_experiments" type="stop_simulation_node" name="stop_simulation_node" output="screen" required="true">
        <param name="data_log_folder" value="$(arg log_path)/$(arg nav_profile)_"/>
        <param name="store_data_freq" value="$(arg store_data_freq)" type="double"/>
        <param name="increase_power_factor" value="$(arg increase_power)" type="double"/>
        <param name="send_laser_error" value="$(arg send_laser_error)" type="bool"/>
        <param name="nfr_energy" value="$(arg nfr_energy)"/>
        <param name="nfr_safety" value="$(arg nfr_safety)"/>
    </node>

    
    <!-- Record data  -->
    <node pkg="rosbag" type="record" name="record_bag_node" output="screen" if="$(arg record_bags)"
        args="-o $(arg bag_store_path)/metacontrol_
        /amcl_pose
        /power_load
        /odom
        /imu/data
        /d_obstacle
        /diagnostics
        /move_base/result
        /move_base/goal"
        required="true">
    </node>
    <node pkg="metacontrol_experiments" type="create_unknown_obstacles.py" name="create_unknown_obstacles_node" output="screen">
        <param name="goal_nr" value="$(arg goal_nr)" type="int"/>
        <param name="n_obstacles" value="$(arg obstacles)" type="int"/>
    </node>

    <node pkg="cob_monitoring" type="cpu_monitor.py" name="cpu_monitoring" output="screen">
    </node>



</launch>
